File: mycode/catkin_ws/src/my_robot_tutorials/scripts/number_counter.py
#!/usr/bin/env python

import rospy
from std_msgs.msg import Int64
from std_srvs.srv import SetBool

counter = 0;

def handle_reset_number_count(req):
	global counter
	counter = 0
	rospy.loginfo("Counter reset")
	success = True
	return success

def callback_receive_number(msg):
	global counter 
	counter = counter+msg.data

	rospy.loginfo("Number received: ")
	rospy.loginfo(msg)
	pub = rospy.Publisher('/number_count', Int64, queue_size = 10)
	num = Int64()
	num.data = counter
	pub.publish(num)

if __name__ == '__main__':
	rospy.init_node('number_counter_py', anonymous=True) 

	sub = rospy.Subscriber('/number',Int64, callback_receive_number)
	service = rospy.Service('/reset_number_count',SetBool,handle_reset_number_count)
	rospy.spin()

File: mycode/catkin_ws/src/my_robot_tutorials/src/number_counter.cpp
#include <ros/ros.h>
#include <std_msgs/Int64.h>
#include <std_srvs/SetBool.h>

int counter = 0;
ros::Publisher pub;

bool handle_reset_number_counter(std_srvs::SetBool::Request &req,
								std_srvs::SetBool::Response &res) {
	ROS_INFO("Reset Counter");
	counter = 0;
	res.success = true;
	return true;
}

void callback_receive_number(const std_msgs::Int64& msg){
	counter +=msg.data;
	std_msgs::Int64 num;
	num.data = counter;
	pub.publish(num);
}

int main(int argc, char **argv){
	ros::init(argc,argv,"number_coutner_cpp",ros::init_options::AnonymousName);
	ros::NodeHandle nh;

	ros::Subscriber sub = nh.subscribe("/number", 1000, callback_receive_number);
	ros::ServiceServer server = nh.advertiseService("/reset_number_count",handle_reset_number_counter);
	pub = nh.advertise<std_msgs::Int64>("/number_count", 10);

	ros::spin();
}